File: core/src/net/natruid/jungle/components/AttributesComponent.java
package net.natruid.jungle.components;

import com.artemis.Component;
import com.artemis.annotations.PooledWeaver;
import net.natruid.jungle.utils.types.AttributeType;

@PooledWeaver
public class AttributesComponent extends Component {
    public final int[] base = new int[AttributeType.length];
    public final int[] modified = new int[AttributeType.length];
    public boolean dirty = true;

    public int get(AttributeType attribute) {
        return modified[attribute.ordinal()];
    }

    protected void reset() {
        for (int i = 0; i < AttributeType.length; i++) {
            base[i] = 10;
            modified[i] = 10;
        }
    }
}

File: core/src/net/natruid/jungle/components/AttributesComponent.kt
package net.natruid.jungle.components

import com.artemis.Component
import net.natruid.jungle.utils.AttributeType

class AttributesComponent : Component() {
    inline val intelligence get() = modified[AttributeType.INTELLIGENCE.ordinal]
    inline val determination get() = modified[AttributeType.DETERMINATION.ordinal]
    inline val endurance get() = modified[AttributeType.ENDURANCE.ordinal]
    inline val awareness get() = modified[AttributeType.AWARENESS.ordinal]
    inline val luck get() = modified[AttributeType.LUCK.ordinal]

    val base = Array(AttributeType.size) { 10 }
    val modified = Array(AttributeType.size) { 10 }

    var dirty = true
}
